tests/models/test_restore.py

```diff
@@ -60,8 +60,7 @@ def test_resume_from_checkpoint(tmpdir):
         default_root_dir=tmpdir,
         max_epochs=2,
         logger=False,
-        checkpoint_callback=checkpoint_callback,
-        callbacks=[ModelTrainerPropertyParity()]  # this performs the assertions
+        callbacks=[checkpoint_callback, ModelTrainerPropertyParity()]  # this performs the assertions
```

Comment:
I think this test is quite poor. Can you expand it to check that the `Trainer(resume_from_checkpoint=...)` checkpoint was restored correctly? Also that there are no extra ModelCheckpoints to avoid a regression of #4027

Reply:
yes of course, I will do that. thanks for the suggestion. 
I'm currently waiting for a decision by core about the api change. we are currently discussing this.
if this is accepted, I will finish this PR properly and we can deal with the details.

Reply:
Thanks a lot for your time!

Regardless of the decision, I think #4027 is a high priority bug to fix. I don't know what alternatives are being discussed but this one seems sound :rocket:

Reply:
True, I agree. I definitely want to get this done this week.

Reply:
@carmocca so core team agrees with the changes. feel free to give me a tough review :)) 
I decided not to touch this test here too much because it was for a bugfix and only aims at testing the model properties. 
Instead, I added a new test that asserts the callbacks state is reloaded properly.